lib/syslog/writer.go

```diff
@@ -241,18 +244,27 @@ func (c bufferedConn) Close() error                { return c.conn.Close() }
 func (c bufferedConn) Flush() error                { return c.buf.Flush() }
 func (c bufferedConn) Write(b []byte) (int, error) { return c.buf.Write(b) }
 
-func dialWriter(network string, address string, config *tls.Config) (w io.Writer, err error) {
+func dialWriter(network string, address string, config *tls.Config, socksProxy string) (w io.Writer, err error) {
 	var conn net.Conn
 	var dial func(string, string) (net.Conn, error)
+	var socksDialer *SocksDialer
 
 	if network == "tls" {
-		network, dial = "tcp", func(network string, address string) (net.Conn, error) {
+		network, dial = "tcp", func(network, address string) (net.Conn, error) {
 			return tls.Dial(network, address, config)
 		}
 	} else {
 		dial = net.Dial
 	}
 
+	if socksProxy != "" {
+		socksDialer = &SocksDialer{tlsConfig: config}
+		if err = socksDialer.connect(network, socksProxy); err != nil {
+			return
+		}
```

Comment:
yeah, I had the feeling like this SocksDialer thing was a bad idea. I've simplified the code ;)